dnsdist: warnings with g++ 12.1.0 on Arch

<!-- Hi! Thanks for filing an issue. It will be read with care by human beings. Can we ask you to please fill out this template and not simply demand new features or send in complaints? Thanks! -->
<!-- Also please search the existing issues (both open and closed) to see if your report might be duplicate -->
<!-- Please don't file an issue when you have a support question, send support questions to the mailinglist or ask them on IRC (https://www.powerdns.com/opensource.html) -->
<!-- Before filing your ticket, please read our 'out in the open' support policy at https://blog.powerdns.com/2016/01/18/open-source-support-out-in-the-open/ -->

<!-- Tell us what is issue is about -->
 - Program: dnsdist <!-- delete the ones that do not apply -->
 - Issue type: Bug report

### Short description
<!-- Explain in a few sentences what the issue/request is -->
```
dnsdist-lua-inspection.cc: In function ‘LuaArray<std::vector<boost::variant<std::__cxx11::basic_string<char>, double> > > getGenResponses(uint64_t, boost::optional<int>, std::function<bool(const Rings::Response&)>)’:
dnsdist-lua-inspection.cc:82:27: warning: narrowing conversion of ‘(count ++)’ from ‘uint64_t’ {aka ‘long unsigned int’} to ‘int’ [-Wnarrowing]
   82 |       ret.push_back({count++, {rc.second.toString(), rc.first, 100.0*rc.first/total}});
      |                      ~~~~~^~
dnsdist-lua-inspection.cc:87:20: warning: narrowing conversion of ‘count’ from ‘uint64_t’ {aka ‘long unsigned int’} to ‘int’ [-Wnarrowing]
   87 |     ret.push_back({count, {"Rest", rest, 100.0*rest/total}});
      |                    ^~~~~
dnsdist-lua-inspection.cc:90:20: warning: narrowing conversion of ‘count’ from ‘uint64_t’ {aka ‘long unsigned int’} to ‘int’ [-Wnarrowing]
   90 |     ret.push_back({count, {"Rest", rest, 100.0 }});
      |                    ^~~~~
  CXX      dnsdist-protobuf.o

dnsdist-rings.cc: In member function ‘std::unordered_map<int, std::vector<boost::variant<std::__cxx11::basic_string<char>, double> > > Rings::getTopBandwidth(unsigned int)’:
dnsdist-rings.cc:120:24: warning: narrowing conversion of ‘(count ++)’ from ‘unsigned int’ to ‘int’ [-Wnarrowing]
  120 |       ret.insert({count++, {rc.second.toString(), rc.first, 100.0*rc.first/total}});
      |                   ~~~~~^~
dnsdist-rings.cc:125:17: warning: narrowing conversion of ‘count’ from ‘unsigned int’ to ‘int’ [-Wnarrowing]
  125 |     ret.insert({count, {"Rest", rest, 100.0*rest/total}});
      |                 ^~~~~
dnsdist-rings.cc:128:17: warning: narrowing conversion of ‘count’ from ‘unsigned int’ to ‘int’ [-Wnarrowing]
  128 |     ret.insert({count, {"Rest", rest, 100.0 }});
```

--- a/pdns/dnsdist-lua-inspection.cc
+++ b/pdns/dnsdist-lua-inspection.cc
@@ -72,10 +72,10 @@ static LuaArray<std::vector<boost::variant<string,double>>> getGenResponses(uint
 
   LuaArray<vector<boost::variant<string,double>>> ret;
   ret.reserve(std::min(rcounts.size(), static_cast<size_t>(top + 1U)));
-  uint64_t count = 1;
+  int count = 1;
   unsigned int rest = 0;
   for (const auto& rc : rcounts) {
-    if (count == top+1) {
+    if (count == static_cast<int>(top + 1)) {
       rest+=rc.first;
     }
     else {

--- a/pdns/dnsdist-rings.cc
+++ b/pdns/dnsdist-rings.cc
@@ -111,9 +111,9 @@ std::unordered_map<int, vector<boost::variant<string,double>>> Rings::getTopBand
 	       });
   std::unordered_map<int, vector<boost::variant<string,double>>> ret;
   uint64_t rest = 0;
-  unsigned int count = 1;
+  int count = 1;
   for(const auto& rc : rcounts) {
-    if(count==numentries+1) {
+    if (count == static_cast<int>(numentries + 1)) {
       rest+=rc.first;
     }
     else {

